core/include/cubos/core/ecs/component/manager.hpp

```diff
@@ -187,18 +188,16 @@ namespace cubos::core::ecs
             std::unique_ptr<std::shared_mutex> mutex; ///< Read/write lock for the storage.
         };
 
-        /// @brief Maps component types to component IDs.
-        std::unordered_map<std::type_index, std::size_t> mTypeToIds;
-
-        std::vector<Entry> mEntries; ///< Registered component storages.
+        memory::TypeMap<std::size_t> mTypeToIds; ///< Maps component types to component IDs.
+        std::vector<Entry> mEntries;             ///< Registered component storages.
     };
 
     // Implementation.
 
     template <typename T>
```

Comment:
Why don't we enforce the concept in templates like these?

Reply:
We should, but these things will all be refactored in a later PR as to not clutter this one.
The ECS still uses `std::type_index` in some places, here I just did the bare minimum changes to make it compile.